starters/nuxt-2/store/checkout.js

```diff
@@ -0,0 +1,34 @@
+import { get, set, del } from 'idb-keyval';
+import createShopifyCheckoutClient from '@nacelle/shopify-checkout';
+
+export const actions = {
+  async initCheckout({ commit }) {
+    try {
+      const checkoutClient = createShopifyCheckoutClient(this.$config.shopify);
+      const checkoutId = await get('checkoutId');
+      if (checkoutId) {
+        const checkout = checkoutClient.get({
+          id: checkoutId
+        });
+        if (checkout?.completed) {
+          await del('checkoutId');
+          commit('cart/clearCart', false, { root: true });
+        }
+      }
+    } catch {}
+  },
+  async processCheckout({ rootState }) {
+    try {
+      const checkoutClient = createShopifyCheckoutClient(this.$config.shopify);
+      const cartItems = rootState.cart.lineItems.map((lineItem) => ({
+        quantity: lineItem.quantity,
+        variantId: lineItem.variant.id
+      }));
+      const checkoutData = await checkoutClient.process({ cartItems });
+      await set('checkoutId', checkoutData.id);
+      if (checkoutData.url) {
+        window.location.href = checkoutData.url;
+      }
+    } catch {}
```

Comment:
We should either:
- Do something in the `catch` block, or:
- Remove the `try/catch`, and add a `.catch` to `.process`, and do something in the `.catch`.

When `catch`ing an error in `checkoutClient.process`, we could at least `console.error(err)`. The merchant developer may want to trigger a modal,  but I think it would be too heavy-handed/opinionated to implement that here (please feel free to disagree!).